File: Homework/Homework2/bad.py
def is_bad_version(x):
    if x >= 15:
        return True
    return False


def first_bad_version(n):
    start = 1
    end = n

    while start < end:
        middle = start + (end - start) // 2

        if is_bad_version(middle):
            end = middle

        else:
            start = middle + 1

    return start


if __name__ == "__main__":
    n = 20
    print(first_bad_version(n))

File: Homework/Homework2/Bad.java
package hw2;

public class Bad {
	
	public static boolean isBadVersion(int x) {
		
		if (x >= 15) {
			return true;
		}
		return false;
		
	}
	
	public static int firstBadVersion(int n) {
		
		int start = 1; 
		int end = n; 
		int middle = 0;
		 
		
		while(start < end) {
			middle = start +(end - start)/2;
			
			
			if(isBadVersion(middle)) {
				 end = middle; 
	
				
			}
			else {
				start = middle +1;
			}
			
			
			
		}
		return start;
	
		
	}
	
	public static void main (String[] args) {
		int n = 20; 
		System.out.print(firstBadVersion(n));
		
	}
	
	
	
	
	
	
	
	

}
